.github/workflows/rust.yml

```
@@ -1,121 +1,88 @@
-name:                  Test Suite
+name:                              Check, test and build Suite
```

```suggestion
name:                              Compilation and Testing Suite
```

This one's a bit more of a nitpick since there's a stage called "Check, test and build" and as an Actions newbie I want to differentiate between them